.github/workflows/validate-crd.yaml

```diff
@@ -0,0 +1,45 @@
+name: Check CRD version update
+
+on:
+  pull_request:
+    paths:
+      - 'pkg/k8s/apis/cilium.io/client/crds/v1alpha1/*.yaml'
+      - 'pkg/k8s/apis/cilium.io/v1alpha1/register.go'
+
+jobs:
+  check-version:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check for CRD changes and version update
+        run: |
+          crd_changed=0
+          version_changed=0
+
+          # Check for CRD changes
+          crd_changes=$(git diff --name-only ${{ github.base_ref }} ${{ github.head_ref }} -- pkg/k8s/apis/cilium.io/client/crds/v1alpha1/*.yaml)
+          if [ -n "$crd_changes" ]; then
+            crd_changed=1
+          fi
+
+          # Check for version variable changes
+
+          old_version=$(git show ${{ github.event.pull_request.base.sha }}:pkg/k8s/apis/cilium.io/v1alpha1/register.go | grep 'CustomResourceDefinitionSchemaVersion' | awk -F'"' '{print $2}')
+          new_version=$(git show HEAD:pkg/k8s/apis/cilium.io/v1alpha1/register.go | grep 'CustomResourceDefinitionSchemaVersion' | awk -F'"' '{print $2}')
```

Comment:
Not sure how this would work, since `CustomResourceDefinitionSchemaVersion` lives `version.go`:
https://github.com/cilium/tetragon/blob/03beec7f1a0ca06675ff4bf7d89fec9307db8359/pkg/k8s/apis/cilium.io/v1alpha1/version.go#L10.

Not sure why my original example had `register.go` (maybe it was based on an older version), but please ensure that everything works properly in the current `main` branch.

Thanks!

Reply:
Thanks I changed and tested I get the values 

```
[Check CRD version update/check-version]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2] user= workdir=
| old_version=1.1.1
| new_version=1.1.2
| Version updated but CRD not changed
```